sunpy/util/tests/test_util.py

```diff
@@ -48,11 +53,23 @@ def test_expand_list():
     lst = [1, 2, 3, [4, 5, 6], 7, (8, 9), ((10, 11), ((12, 13),))]
     assert util.expand_list(lst) == [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13]
 
-
-def test_expand_list_generator():
-    lst = ['a', 'b', [], (['c', 'd']), tuple(), ['e']]
-    assert list(util.expand_list_generator(lst)) == ['a', 'b', 'c', 'd', 'e']
-
+@pytest.mark.filterwarnings("ignore:Invalid 'BLANK' keyword in header")
+def test_expand_list_generator(aia171_test_map):
+    # We want to ensure that it does not expand:
+    # - strings, bytes, array-like objects, WCS or FITS headers
+    # Only lists, tuples and Generator objects should be expanded
+    header = fits.getheader(get_test_filepath('aia_171_level1.fits'))
+    generator = iter(['c', 'd'])
+    zip_generator = zip(['1', '2'], ['3', '4'])
+    first_list = ['a1234', 'b', [], (['c', 'd']), tuple(), ['e'], b'fghj']
+    # The empty list and tuple should be ignored
+    assert list(util.expand_list_generator(first_list)) == ['a1234', 'b', 'c', 'd', 'e', b'fghj']
+    assert list(util.expand_list_generator([aia171_test_map.data, aia171_test_map.wcs, header])) == [aia171_test_map.data, aia171_test_map.wcs, header]
+    assert list(util.expand_list_generator(generator)) == ['c', 'd']
+    assert list(util.expand_list_generator(zip_generator)) == ['1', '3', '2', '4']
+    with open(get_test_filepath('aia_171_level1.fits'), 'rb') as f:
+        with open(get_test_filepath('aia_171_level1.fits'), 'rb') as f2:
+            assert list(util.expand_list_generator([f])) == f2.readlines()
```

Comment:
This is what I came up with, I hate it.

Reply:
Isn't this a more efficient way of doing the same thing 
```
with open(get_test_filepath('aia_171_level1.fits'), 'rb') as f:
        file_contents = f.readlines()
        assert list(util.expand_list_generator([f])) == file_contents
```

Reply:
I thought that when you call readlines, it consumes f, so it's now at the end of the file. 

I guess I can just rewind it?